### PR TITLE
Set the VMDq pool size correctly

### PR DESCRIPTION
**Pull request by stephen-hurd**
Needs to be the minimum of all used resources... vnic, l2 CTX, RSS
CTX, or 64 (hard coded).

### CODE REVIEW

**Comment by ajitkhaparde** [comment]
We probably don't have max_stat_ctx enough for Tx and Rx rings combined. Do we?

**Comment by stephen-hurd** [comment]
I haven't seen a stat context allocation fail yet...

File: drivers/net/bnxt/bnxt_rxq.c [hunk 1]
```diff
@@ -122,8 +122,14 @@ int bnxt_mq_rx_configure(struct bnxt *bp)
...
-				"VMDq pool not set, defaulted to 64\n");
-			pools = ETH_64_POOLS;
+			    "VMDq pool not set, defaulted to %d\n",
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
+			pools = RTE_MIN(bp->max_vnics,
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_rxq.c]: Should we check against max_rsscos_ctx only when RSS flag is set?
Also do we need to export the right max_vmdq_pools/rings in dev_get_info?

File: drivers/net/bnxt/bnxt_rxq.c [hunk 2]
```diff
@@ -122,8 +122,14 @@ int bnxt_mq_rx_configure(struct bnxt *bp)
...
 		if (!pools) {
 			RTE_LOG(ERR, PMD,
-				"VMDq pool not set, defaulted to 64\n");
-			pools = ETH_64_POOLS;
+			    "VMDq pool not set, defaulted to %d\n",
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
+			pools = RTE_MIN(bp->max_vnics,
+			    RTE_MIN(bp->max_vnics,
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_rxq.c]: max_vnics again? copy-paste?

File: drivers/net/bnxt/bnxt_rxq.c (hunk 1 above)
Comment (stephen-hurd) [reply drivers/net/bnxt/bnxt_rxq.c]: I think we always need the limit.
Yeah, I do think something is needed in dev_info_get_op()...

**Review by ajitkhaparde — APPROVED** [review]

**Pull request merged**